rcl/src/rcl/arguments.c

```diff
@@ -362,32 +362,34 @@ rcl_remove_ros_arguments(
   int * nonros_argc,
   const char ** nonros_argv[])
 {
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(argv, RCL_RET_INVALID_ARGUMENT, allocator);
   RCL_CHECK_ARGUMENT_FOR_NULL(nonros_argc, RCL_RET_INVALID_ARGUMENT, allocator);
   RCL_CHECK_ARGUMENT_FOR_NULL(args, RCL_RET_INVALID_ARGUMENT, allocator);
-  RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
 
   *nonros_argc = rcl_arguments_get_count_unparsed(args);
+  *nonros_argv = NULL;
+
   if (*nonros_argc > 0) {
     int * unparsed_indices = NULL;
     rcl_ret_t ret;
     ret = rcl_arguments_get_unparsed(args, allocator, &unparsed_indices);
 
     if (RCL_RET_OK != ret) {
-      allocator.deallocate(unparsed_indices, allocator.state);
-      unparsed_indices = NULL;
       return ret;
     }
 
-    *nonros_argv = allocator.allocate(sizeof(char *) * *nonros_argc, allocator.state);
+    size_t alloc_size = sizeof(char *) * *nonros_argc;
+    *nonros_argv = allocator.allocate(alloc_size, allocator.state);
+    if (NULL == *nonros_argv) {
+      return RCL_RET_BAD_ALLOC;
```

Comment:
Do you leak `unparsed_indices` here?

Reply:
Yes.

Also as a question, does deallocate need to be checked as well?

Reply:
> Also as a question, does deallocate need to be checked as well?

No, it doesn't have a return code. Things like double-free usually result in an abort.